### PR TITLE
Handle NotImplementedError when creating symlink on install; fixes #4626

### LINKED ISSUE #4626: symlink can raise NotImplementedError
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [ ] If an exception occurs when executing a command, I executed it again in debug mode (`-vvv` option).

- **OS version and name**: windows
- **Poetry version**: HEAD
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: 

## Issue

PyPy does not yet implement os.symlink on windows. If poetry wishes to support PyPy on windows, this line needs to also trap NotImplementedError https://github.com/python-poetry/poetry/blob/656c71d46f5cd35ac7996b00c631e5056c883b55/install-poetry.py#L561

### PR DESCRIPTION
**Pull request by havocbane**
Handle `NotImplementedError` when creating symlink on install. I was able to recreate the issue locally with PyPy version 3.7.10 and Poetry version 1.1.11 and I verified my fix after making changes: `python install-poetry.py` completed successfully with this change in place. If accepted, please tag the merge with the label **hacktoberfest-accepted**.

Resolves: #4626 

- [ ] Added **tests** for changed code.
- [X] Updated **documentation** for changed code.

I am happy to add a test over this change if requested, but I don't see where tests covering the `install-poetry.py` script live in the repository.

### CODE REVIEW

**Comment by mattip** [comment]
I don't know if the poetry CI is already taking too much time, but you could add  runs for python3.10 and pypy3.7 by changing
https://github.com/python-poetry/poetry/blob/82459bd6f38785882a8eff0fbce2a8b0119b205c/.github/workflows/main.yml#L26

to (note the quotes to force a string, not a number, critical for 3.10)
``` 
python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "'pypy-3.7']
```

**Comment by havocbane** [comment]
> I don't know if the poetry CI is already taking too much time, but you could add runs for python3.10 and pypy3.7 by changing
> 
> https://github.com/python-poetry/poetry/blob/82459bd6f38785882a8eff0fbce2a8b0119b205c/.github/workflows/main.yml#L26
> 
> to (note the quotes to force a string, not a number, critical for 3.10)
> 
> ```
> python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "'pypy-3.7']
> ```

Thanks for the suggestion. The CI is just waiting for a maintainer to kick it off since I have not contributed to this codebase before. Also, since this change is focused on pypy, I will omit 3.10 from the update.

**Comment by havocbane** [comment]
I'm not sure why the FreeBSD checks are failing installing tox. Could it be due to the pinning of the PyPi version? It looks unrelated to me.

**Comment by abn** [comment]
@havocbane try rebasing the branch, #4668 should remove tox usage.

**Comment by havocbane** [comment]
> @havocbane try rebasing the branch, #4668 should remove tox usage.

Thanks! The FreeBSD checks are passing now.

**Comment by abn** [comment]
@havocbane can you resolve the install-poetry conflicts please?

File: poetry/repositories/pypi_repository.py
Comment (neersighted) [inline poetry/repositories/pypi_repository.py:325]: Looks like some unrelated changes came along for the ride. Could we back these out, and open a separate PR with tests if they are necessary?

**Review by neersighted — CHANGES_REQUESTED** [review]

**Comment by neersighted** [comment]
Given the extra changes that came along, and the fact that the installer script has been factored out, I am going to close this in favor of a PR at https://github.com/python-poetry/install.python-poetry.org.

For anyone using the old URL, they will not get the improved version of this script after merge. However, I would prefer an issue on that repo to discuss if we'll keep the version in the main poetry repository in sync, or something else.

File: poetry/repositories/pypi_repository.py
Comment (havocbane) [reply poetry/repositories/pypi_repository.py:325]: Ah, good catch. I was trying to debug the mac failures from the CI. I'll back these out in a new pull request.

**Comment by hoefling** [comment]
I encountered the same issue with being unable to install `poetry` on win + pypy3. @havocbane was there a reason you didn't port this PR to https://github.com/python-poetry/install.python-poetry.org?

**Comment by havocbane** [comment]
> I encountered the same issue with being unable to install `poetry` on win + pypy3. @havocbane was there a reason you didn't port this PR to https://github.com/python-poetry/install.python-poetry.org?

@hoefling Unfortunately, life kind of got in the way and I didn't keep up with the changes here. If this is still an issue, I can make a change over there. I know a lot of time has passed since this came up, so sorry for that.

**Comment by mattip** [comment]
> I encountered the same issue with being unable to install poetry on win + pypy3

What version of pypy3? The latest pypy7.3.13 added support for symlinks in windows, so it should work now.

**Pull request closed without merging**
